### ISSUE TITLE
HEDA Setting (option) to keep 'affiliation' history when a 'program enrollment' is deleted and a new one is created

### ISSUE THREAD

**Issue by aligeez** [SalesforceFoundation/EDA#704]
* Feature Enhancement 

* **Current Behavior** 
A Program Enrollment is deleted (and the Affiliation is not deleted but is marked "Former") but then a Program Enrollment is added back. New values do not overwrite the old values.

* **Enhancement Behavior?**
Admins will have the option to keep affiliation history when a program enrollment is deleted. When a new Program Enrollment is created it would create a new affiliation. This way the Advisor (for example) can see historically that Sophia Student has changed her major 4 times in the past 3 semesters - this history persists. Also useful if a student has dropped out and re-enrolls, that may be something an educational institution would like to keep track of. 

Default Behavior: one customer may want to just leverage the same Affiliation and re-update it to current and copy the new dates. 

However, I can see some places wanting that history, in that if they leave but come back they'd want to see both affiliations (< -- new option/setting). 

* **Potential Future Functionality**
Basically, search for an affiliation, if one is found but has a status of Former, either update the status, add dates; or create a new one.

* **Acceptance Criteria**
* Admin goes to settings at selects one or the other option described above.
* If option 1 is selected (default behavior), a contact that has a PE deleted, and then a new PE created, it will utilize the existing Affiliation and update the relevant fields. 
* If option 2 is selected, a contact has a PE deleted, the affiliation stays in a 'former' state, and when a new PE is created it also creates a new affiliation; essentially showing a history of their former affiliations.

**Comment by aligeez** [comment]
**lurch: add

**Comment by aligeez** [comment]
Closing this issue as we've discussed it across the Scrum team and it's not something we need to implement in EDA.
